src/hotspot/cpu/s390/s390.ad

```diff
@@ -540,6 +733,25 @@ reg_class z_dbl_reg(
 );
 reg_class z_rscratch1_dbl_reg(Z_F1,Z_F1_H);
 
+reg_class z_v_reg(
+ Z_VR16, Z_VR16_H, Z_VR16_J, Z_VR16_K,
+ Z_VR17, Z_VR17_H, Z_VR17_J, Z_VR17_K,
+ Z_VR18, Z_VR18_H, Z_VR18_J, Z_VR18_K,
+ Z_VR19, Z_VR19_H, Z_VR19_J, Z_VR19_K,
+ Z_VR20, Z_VR20_H, Z_VR20_J, Z_VR20_K,
+ Z_VR21, Z_VR21_H, Z_VR21_J, Z_VR21_K,
+ Z_VR22, Z_VR22_H, Z_VR22_J, Z_VR22_K,
+ Z_VR23, Z_VR23_H, Z_VR23_J, Z_VR23_K,
+ Z_VR24, Z_VR24_H, Z_VR24_J, Z_VR24_K,
+ Z_VR25, Z_VR25_H, Z_VR25_J, Z_VR25_K,
+ Z_VR26, Z_VR26_H, Z_VR26_J, Z_VR26_K,
+ Z_VR27, Z_VR27_H, Z_VR27_J, Z_VR27_K,
+ Z_VR28, Z_VR28_H, Z_VR28_J, Z_VR28_K,
+ Z_VR29, Z_VR29_H, Z_VR29_J, Z_VR29_K,
+ Z_VR30, Z_VR30_H, Z_VR30_J, Z_VR30_K,
+ Z_VR31, Z_VR31_H, Z_VR31_J, Z_VR31_K,
+);
+
```

Comment:
Why do you use only VR16 to VR31? If there is a reason, it should be explained in a comment.

Reply:
In s390, 1st 16 VRs are aliases for the FPRs same as PPC. I will add proper explanation for this.

Reply:
You use `ALLOC_IN_RC(z_v_reg)` in `vecX(),` so FP and VR are disjoint. That looks correct. Only the comments are missing.

Reply:
Thanks @TheRealMDoerr , I will add comments.